pkg/provisioner/cloudprovisioner.go

```diff
@@ -645,10 +657,10 @@ func (c *CloudProvisioner) PublishVolume(
 			klog.Fatalf("failed to extract scaleset name: %v", err)
 		}
 
-		vmEntry, err := c.cloud.Get(ctx, fullScaleSetName, attachedNode, instanceID)
+		vmEntry, err := c.cloud.GetVMSSVM(ctx, attachedNode)
 		klog.Infof("vmEntry: %+v", vmEntry)
-		storageProfile := vmEntry.StorageProfile
-		vmState = vmEntry.ProvisioningState
+		storageProfile := vmEntry.VM.Properties.StorageProfile
```

Comment:
Either add a null check here to ensure that the Properties or StorageProfile is not null or add a function in the cache to return the StorageProfile and error to handle any error scenario.